includes/class-wc-payments-payment-request.php

```diff
@@ -0,0 +1,1232 @@
+<?php
+/**
+ * Stripe Payment Request API
+ * Adds support for Apple Pay and Chrome Payment Request API buttons.
+ * Utilizes the Stripe Payment Request Button to support checkout from the product detail and cart pages.
+ *
+ * Adapted from WooCommerce Stripe Gateway extension.
+ *
+ * @package WooCommerce\Payments
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+use WCPay\Logger;
+
+/**
+ * WC_Payments_Payment_Request class.
+ */
+class WC_Payments_Payment_Request {
+	/**
+	 * Gateway settings.
+	 *
+	 * @var array
+	 */
+	public $gateway_settings;
+
+	/**
+	 * Total label
+	 *
+	 * @var string
+	 */
+	public $total_label;
+
+	/**
+	 * Is test mode active?
+	 *
+	 * @var bool
+	 */
+	public $testmode;
+
+	/**
+	 * This Instance.
+	 *
+	 * @var WC_Payments_Payment_Request
+	 */
+	private static $instance;
+
+	/**
+	 * WC_Payments_Account instance to get information about the account
+	 *
+	 * @var WC_Payments_Account
+	 */
+	private $account;
+
+	/**
+	 * Initialize class actions.
+	 *
+	 * @param WC_Payments_Account $account Account information.
+	 */
+	public function __construct( WC_Payments_Account $account ) {
+		self::$instance = $this;
+
+		// Add default settings in case the payment request options are missing.
+		$this->gateway_settings = array_merge( self::get_default_settings(), get_option( 'woocommerce_woocommerce_payments_settings', [] ) );
+		$this->account          = $account;
+		$this->testmode         = ( ! empty( $this->gateway_settings['test_mode'] ) && 'yes' === $this->gateway_settings['test_mode'] ) ? true : false;
+		$this->total_label      = ! empty( $this->account->get_statement_descriptor() ) ? $this->account->get_statement_descriptor() : '';
```

Comment:
Thanks for digging into this 🙏

I've taken that out of the constructor and put it into a function. I also refactored the code a little bit, because this if `! empty( $this->account->get_statement_descriptor() ) ?` is not necessary since `get_statement_descriptor()` already returns an empty string in case the value is not set.